src/inference/src/dev/core_impl.cpp

```diff
@@ -1289,7 +1289,7 @@ void ov::CoreImpl::set_property_for_device(const ov::AnyMap& configMap, const st
                 if (device_supports_cache_dir(plugin.second)) {
                     ov::AnyMap empty_map;
                     auto cacheConfig = coreConfig.get_cache_config_for_device(plugin.second, empty_map);
-                    if (cacheConfig._cacheManager) {
+                    if (cacheConfig._cacheManager || cacheConfig._unsetCacheDir) {
```

Comment:
what problem we are trying to solve?

Reply:
https://jira.devtools.intel.com/browse/CVS-132745